timemachine/fe/energy_decomposition.py

```diff
@@ -0,0 +1,80 @@
+import functools
+from dataclasses import dataclass
+from typing import Callable, List, Sequence
+
+import numpy as np
+
+from timemachine.constants import BOLTZ, DEFAULT_TEMP
+
+Frames = np.ndarray
+Boxes = np.ndarray
+ReducedEnergies = np.ndarray
+Batch_u_fn = Callable[[Frames, Boxes], ReducedEnergies]
+
+
+@dataclass
+class EnergyDecomposedState:
+    """contains samples (frames, boxes) and a list of reduced energy functions"""
+
+    frames: np.ndarray
+    boxes: np.ndarray
+    batch_u_fns: Sequence[Batch_u_fn]  # u_fn : (frames, boxes) -> reduced_energies
+
+
+def get_batch_u_fns(bps, temperature=DEFAULT_TEMP):
+    # return a list of functions that take in (coords, boxes), return reduced_potentials
```

Comment:
Use a docstring rather then an inline comment?

Reply:
Updated in https://github.com/proteneer/timemachine/pull/936/commits/831f0035a93560b630a785fd512f115b6533048e